components/SuggestionCard.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+  The suggestion card is used as part of the SearchPanel. It displays
+  the autocomplete inputs.
+*/
+
+import {
+  Text,
+  Container,
+  Heading,
+  HStack,
+  Stack,
+  Image,
+  Skeleton,
+} from "@chakra-ui/react";
+import { useBookQuery } from "@lib/hooks/book";
+import { CourseWithDept } from "@lib/services/course";
+import { Book } from "@prisma/client";
+
+const fallbackImageSrc =
+  "https://campusstore.mcmaster.ca/cgi-mcm/ws/getTradeImage.pl?isbn=281000000883B";
+
+export interface SuggestionProps {
+  type: string;
+  entry: Book | CourseWithDept;
+}
+
+export const CourseSuggestionCard = ({ name, code, dept }: CourseWithDept) => {
+  return (
+    <HStack>
+      <Stack>
+        <Container alignContent={"left"} marginY={2}>
+          <Heading isTruncated as="h4" size="md" color="primaryText">
+            {`${dept.abbreviation} ${code}`}
+          </Heading>
+          <Text color="secondaryText" isTruncated>
+            {name || ""}
+          </Text>
+        </Container>
+      </Stack>
+    </HStack>
+  );
+};
+
+export const BookSuggestionCard = ({ isbn, name, imageUrl }: Book) => {
+  const { isLoading, data: populatedBook } = useBookQuery(isbn);
+  let authorString: string = "-";
+  if (populatedBook?.googleBook?.authors) {
+    authorString = populatedBook.googleBook.authors.join(", ");
+  }
+
+  return (
+    <HStack>
+      <Image
+        maxH={"100px"}
+        maxW={"60px"}
+        src={imageUrl || fallbackImageSrc}
+        alt={"book image"}
+      ></Image>
+      <Stack>
+        <Container alignContent={"left"} marginY={2}>
+          <Heading isTruncated as="h4" size="md" color="primaryText">
+            {`${name}`}
+          </Heading>
+          <Skeleton isLoaded={!isLoading}>
+            <Text color="secondaryText" isTruncated>
+              {authorString}
+            </Text>
+          </Skeleton>
+        </Container>
+      </Stack>
+    </HStack>
+  );
+};
+
+export const SuggestionCard = ({ type, entry }: SuggestionProps) => {
```

Comment:
Consider this instead (in combination with changing the `AutocompleteItem` definition)
```typescript
export interface SuggestionCardProps {
  item: AutocompleteItem
}

export const SuggestionCard: React.FC<SuggestionCardProps> = ({ item }) => {
```

This avoids unnecessarily duplicating the `AutocompleteItem` definition (although it requires stronger coupling but I think that is okay/helpful here). It should also mean you can get rid of the type casts in the component body.